File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        Map<String, Integer> memory = new HashMap<>();
        
        for(int i = 0; i < yearning.length; i++){
            memory.put(name[i], yearning[i]);
        }
        
        for(int j = 0; j < photo.length; j++){
            for(int k = 0; k < photo[j].length; k++){
                if(memory.containsKey(photo[j][k])){
                    int score = memory.get(photo[j][k]);
                    answer[j] += score;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    dic = dict(zip(name,yearning))
    
    for p in photo:
        score = 0
        for x in p:
            if x in dic:
                score += dic[x]    
        answer.append(score)
    
    return answer